core/pipeline.go

```diff
@@ -311,3 +321,23 @@ func (p *BasePipeline) SyncEnvironment(sessionCtx context.Context, sess *Session
 func (p *BasePipeline) Docker() bool {
 	return p.config.Docker
 }
+
+func checkError(exit int, output []string, err error) error {
+	if exit != 0 {
+		return fmt.Errorf("Unable to sync environment, exit code: %d", exit)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if checkOutputEmpty(output) {
+		return fmt.Errorf("No output returned by env command from container")
+	}
+
+	return nil
+}
+
+func checkOutputEmpty(output []string) bool {
+	return !(output != nil && len(output) > 0)
```

Comment:
This is more readable imo:

```suggestion
	return output == nil || len(output) == 0
```

Reply:
Fixed